src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2496,23 +2496,25 @@ bool IsValidIdentifierName(const char *name) {
 mgp_error mgp_message_payload(mgp_message *message, const char **result) {
   return WrapExceptions(
       [message] {
-        return std::visit(
-            utils::Overloaded{[](const mgp_message::KafkaMessage &msg) { return msg->Payload().data(); },
-                              [](const mgp_message::PulsarMessage &msg) { return msg.Payload().data(); },
-                              [](const auto & /*other*/) { throw std::invalid_argument("Invalid source type"); }},
-            message->msg);
+        return std::visit(utils::Overloaded{[](const mgp_message::KafkaMessage &msg) { return msg->Payload().data(); },
+                                            [](const mgp_message::PulsarMessage &msg) { return msg.Payload().data(); },
+                                            [](const auto & /*other*/) -> const char * {
```

Comment:
All the return types of the overloads should have the same return type. Clearly, the third lambda throws, it's a little bit verbose that you have to explicitly specify that, or otherwise, you get a compiler error. Anyway, this is not a comment on the PR itself and definitely, I think overloaded is so much more compact than then `if constexpr` variant.

Reply:
yes it is, that's why I went and changed all of them to that style.
Good thing I only have to define the return type for the last overload that throws. :smile: